### PR TITLE
Make fastlane installation optional

### PR DESCRIPTION
**Pull request by jeffyoung**
- Adding the missing launch.json file I thought I pushed in December.  This allows you to run tests via F5.

- Allowing all 3 tasks to optionally install the fastlane tools.  If the user wants them installed, they can choose to always use the latest or a specific version.

- Updated existing tests to handle new properties on task.  Added new tests to verify new property behavior.

- Version the extension to 114.

### CODE REVIEW

**Comment by jeffyoung** [comment]
This will allow users to choose a version of Fastlane to install which should provide a workaround for #60.  Users can also choose *not* to install Fastlane at all (if they have specific configurations they require).

File: Tasks/app-store-promote/task.json [hunk 1]
```diff
@@ -119,6 +119,38 @@
...
+            "name": "fastlaneToolsVersion",
+            "type": "pickList",
+            "label": "Fastlane Version",
+            "defaultValue": "LatestVersion",
+            "options": {
+                "LatestVersion": "Latest Version",
+                "SpecificVersion": "Specific Version"
+            },
+            "groupName": "advanced",
+            "helpMarkDown": "Choose to install either the lastest version of fastlane or a specific version.",
```
Comment (DavidStaheli) [inline Tasks/app-store-promote/task.json]: Capitalize as "Fastlane" here too?
(Same for other tasks.)

File: Tasks/app-store-promote/task.json [hunk 2]
```diff
@@ -119,6 +119,38 @@
...
+            "visibleRule": "installFastlane = true"
+        },
+        {
+            "name": "fastlaneToolsSpecificVersion",
+            "type": "string",
+            "label": "Fastlane Specific Version",
+            "defaultValue": "",
+            "required": true,
+            "groupName": "advanced",
+            "helpMarkDown": "Provide the version of fastlane to install (e.g., 2.15.1).",
```
Comment (DavidStaheli) [inline Tasks/app-store-promote/task.json]: Capitalize as "Fastlane" here too?
(Same for other tasks.)

File: Tasks/app-store-promote/app-store-promote.ts [hunk 3]
```diff
@@ -65,14 +71,28 @@ async function run() {
...
-        await gemRunner.exec();
+
+        // If desired, install the fastlane tools (if they're already present, should be a no-op)
+        if (installFastlane) {
+            tl.debug('Installing fastlane...');
+            let gemRunner: ToolRunner = tl.tool(tl.which('gem', true));
+            gemRunner.arg(['install', 'fastlane']);
+            if (fastlaneVersionToInstall) {
+                tl.debug(`Installing specific version of fastlane: ${fastlaneVersionToInstall}`);
+                gemRunner.arg(['-v', fastlaneVersionToInstall]);
```
Comment (yacaovsnc) [inline Tasks/app-store-promote/app-store-promote.ts]: What if I already have v1 installed, and I want to install v2?  Would this 'gem install  fastlane -v 2' run and overwrite old fastlane v1?

File: Tasks/app-store-promote/task.json (hunk 1 above)
Comment (madhurig) [inline Tasks/app-store-promote/task.json]: fastlane should actually be lower case - their logo also shows lowercase https://fastlane.tools/

File: Tasks/app-store-promote/app-store-promote.ts (hunk 3 above)
Comment (jeffyoung) [inline Tasks/app-store-promote/app-store-promote.ts]: That's what I've seen.  I've also seen v1 be installed if v2 is already present.

File: Tasks/app-store-promote/task.json (hunk 1 above)
Comment (jeffyoung) [inline Tasks/app-store-promote/task.json]: Lowercase it is.  Thanks for the catch!

File: README.md [hunk 4]
```diff
@@ -103,6 +103,12 @@ Allows you to release updates to your iOS TestFlight beta app or production app
 
 2. **Team Name** *(String)* - The name of the producing team. Only necessary when in multiple teams.
 
+3. **Install Fastlane** *(Checkbox)* - By default, install a version of the Fastlane tools.  Uncheck if your build machine already has the version of Fastlane to use.
```
Comment (madhurig) [inline README.md]: Use lowecase fastlane and provide a link to fastlane tools page

Comment (jeffyoung) [inline README.md]: Will do

**Pull request merged**